sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -69,19 +71,22 @@ public HttpLoggingPolicy(HttpLogDetailLevel detailLevel, boolean prettyPrintJSON
     }
 
     private Mono<Void> logRequest(final ClientLogger logger, final HttpRequest request) {
-        if (detailLevel.shouldLogUrl()) {
+        if (httpLogOptions.getLogLevel().shouldLogUrl()) {
             logger.info("--> {} {}", request.getHttpMethod(), request.getUrl());
         }
 
-        if (detailLevel.shouldLogHeaders()) {
-            for (HttpHeader header : request.getHeaders()) {
-                logger.info(header.toString());
-            }
+        if (httpLogOptions.getLogLevel().shouldLogHeaders()) {
+            formatAllowableHeaders(httpLogOptions.getAllowedHeaderNames(), request, logger);
+        }
+
+        if (httpLogOptions.getLogLevel().shouldLogQueryParams()) {
+            formatAllowableQueryParams(httpLogOptions.getAllowedQueryParamNames(), request, logger);
         }
+
         //
         Mono<Void> reqBodyLoggingMono = Mono.empty();
         //
-        if (detailLevel.shouldLogBody()) {
+        if (httpLogOptions.getLogLevel().shouldLogBody()) {
```

Comment:
Whenever I see a method call (like `getLogLevel()`) multiple times I prefer to see it referred to in a local variable.